pkg/ingress2httpproxy/ingress2httpproxy_test.go

```diff
@@ -0,0 +1,52 @@
+package ingress2httpproxy
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	core "k8s.io/api/networking/v1beta1"
+)
+
+const clientName = "testClient"
+
+func TestBuildHP(t *testing.T) {
+	m := newMutatorFromFileData(t, "example.json")
+	ingress := m.input
+	hp := m.buildHTTPProxy()
+
+	assert.Equal(t, "HTTPProxy", hp.Kind)
+	assert.Equal(t, ingress.Name, hp.Name)
+	assert.Equal(t, "projectcontour.io/v1", hp.APIVersion)
+	assert.Equal(t, ingress.Spec.TLS[0].SecretName, hp.Spec.VirtualHost.TLS.SecretName)
+	assert.Equal(t, ingress.Spec.Rules[0].HTTP.Paths[0].Path, hp.Spec.Routes[0].Conditions[0].Prefix)
+	assert.Equal(t, ingress.Spec.Rules[0].HTTP.Paths[0].Backend.ServiceName, hp.Spec.Routes[0].Services[0].Name)
+	assert.Equal(t, ingress.Spec.Rules[0].HTTP.Paths[0].Backend.ServicePort.IntValue(), hp.Spec.Routes[0].Services[0].Port)
+	assert.Equal(t, ingress.Spec.Rules[0].HTTP.Paths[1].Path, hp.Spec.Routes[1].Conditions[0].Prefix)
+	assert.Equal(t, ingress.Spec.Rules[0].HTTP.Paths[1].Backend.ServiceName, hp.Spec.Routes[1].Services[0].Name)
+	assert.Equal(t, ingress.Spec.Rules[0].HTTP.Paths[1].Backend.ServicePort.IntValue(), hp.Spec.Routes[1].Services[0].Port)
+	//To-do
```

Comment:
for `TODO` items i would suggest making note of whats to be done. looks like when uncommenting this assertion fails. when looking at the output, looks similar to when using a domain that doesn't match the expected format (?)